aldryn_people/models.py

```diff
@@ -71,13 +77,58 @@ def company_description(self):
         )
         return self.safe_translation_getter('description')
 
-    def __str__(self):
-        return self.safe_translation_getter('name', str(self.pk))
-
     class Meta:
         verbose_name = _('Group')
         verbose_name_plural = _('Groups')
 
+    def __str__(self):
+        return self.safe_translation_getter(
+            'name', default=_('Group: {0}').format(self.pk))
+
+    def get_absolute_url(self, language=None):
+        if not language:
+            language = get_current_language()
+        if self.slug:
+            kwargs = {'slug': self.slug}
+        else:
+            kwargs = {'pk': self.pk}
+        with override(language):
+            return reverse('aldryn_people:group-detail', kwargs=kwargs)
+
+    def slugify(self, source_text, i=None):
+        slug = default_slugify(source_text)
+        if i is not None:
+            slug += "_%d" % i
+        return slug
+
+    def save(self, **kwargs):
```

Comment:
I'm not a fan of this logic, to me we shouldn't have fetch all group slugs for every language.